swap_meet/vendor.py

```diff
@@ -1,2 +1,107 @@
+import random
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory=None):
+        # Inventory attribute should be optional. 
+        # If there is no inventory argument was passed into. The default value of inventory will be an empty list.
+        self.inventory = inventory if inventory else []
+    
+    def add(self, new_item):
+        if not new_item:
+            raise ValueError("Item can't be empty.")    
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item):
+        # when there are two or more duplicate items in the inventory, we'll only remove one.
+        try:
+            self.inventory.remove(item)
+            return item       
+        except ValueError:  # When the item is not in inventory will raise value error.
+            return None
+    
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        removed_my_item = self.remove(my_item)
+        removed_their_item = other_vendor.remove(their_item)
+
+        if removed_my_item and removed_their_item:
+            self.add(their_item)
+            other_vendor.add(my_item)
+            return True
+        if removed_my_item:
+            self.add(my_item)
+        if removed_their_item:
+            other_vendor.add(their_item)
+        return False
+
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+        return self.swap_items(other_vendor, my_first_item, their_first_item)
+
+    def get_by_category(self, category):
+        return list(filter(lambda item: item.get_category()== category, self.inventory))
+    
+    def get_best_by_category(self, category):
+        cur_best_item = None
+        cur_best_condition = None
+        for item in self.inventory:
+            if item.get_category() == category:
+                if cur_best_item is None or item.condition > cur_best_condition:
```

Comment:
Okay, this is clean code that only loops through the whole thing once.  In a more complex project, you can probably see how we could use things like `filter` to pull specific sets out of the inventory.  Well done, though!